### PR TITLE
[WIP] Julia code for SparseGrid Generation

### PR DESCRIPTION
**Pull request by wwkong**
The following are available:
- A function to generate isotropic total degree lower sets
- A function to generate Clenshaw-Curtis nodes and weights in **nested** order
- Simple unit tests for verifying the correctness of the code

### CODE REVIEW

File: InterfaceJulia/src/TasOneDimesionalRule.jl [hunk 1]
```diff
@@ -1,7 +1,8 @@
+# A collection of 1D interpolation functions that return points and
+# weights in column major format.
 module TasOneDimensionalRule
 export clenshaw_curtis
 
-# TODO: Make the output incrementally nested.
```
Comment (mkstoyanov) [inline InterfaceJulia/src/TasOneDimesionalRule.jl]: OK, now we can make comments to the Pull-Request. this is good, nothing to change.

File: InterfaceJulia/src/TasData.jl [hunk 2]
```diff
@@ -1,8 +1,7 @@
+# A collection of functions for generating key data structures.
 module TasData
 export create_lower_set, is_itd_elem
 
-# A collection of functions for generating key data structures.
```
Comment (mkstoyanov) [inline InterfaceJulia/src/TasData.jl]: The comment is still useful for the file in general. You can leave it or modify it with something shorter.

**Review by mkstoyanov — CHANGES_REQUESTED** [review]
This is a test for requesting changes on a pull-request.

File: InterfaceJulia/src/TasData.jl (hunk 2 above)
Comment (wwkong) [reply InterfaceJulia/src/TasData.jl]: I will leave it for now as I might add more features to this file later on.

**Comment by mkstoyanov** [comment]
You're doing a lot copy-paste from the C++ code. The goal is to get the grid working without looking at the C++ code so that you can get a better feel of the challenges and to try to find solutions on your own. That will help you put the Tasmanian algorithms in a broader context.

**Comment by wwkong** [comment]
> You're doing a lot copy-paste from the C++ code. The goal is to get the grid working without looking at the C++ code so that you can get a better feel of the challenges and to try to find solutions on your own. That will help you put the Tasmanian algorithms in a broader context.

Gotcha. Although, I think the only logic that I pulled from the C++ code though is TasData.create_lower_set(). the rest of the constructions were built from scratch.

**Comment by mkstoyanov** [comment]
Just testing, no need to merge.

**Pull request closed without merging**